webapp/src/Utils/Utils.php

```diff
@@ -1022,6 +1022,30 @@ public static function generatePassword() : string
         return $rand_str;
     }
 
+    /**
+     * Generate a strong random password
+     *
+     * @return string
+     */
+    public static function generateStrongPassword() : string
```

Comment:
Or just make generatePassword take a length parameter?

Reply:
But generatePassword doesn't support capitals and - and _. Do we want to merge the functions and add an option for that?

Reply:
My personal preference is to keep them as separate functions, but rename the old function to generateShortPassword and call the new function generatePassword / generateStrongPassword.
Maybe also update the documentation to explain that generateShortPassword is intended for passwords distributed to users on paper.
The functions can be implemented separately as they are currently, or refactored e.g. using a new private function that takes a length and set of characters. They are short, so it's not a big deal.
This approach prevents misuse (e.g. providing a short length, or calling the old generatePassword function when the intention is to generate a strong password).

Reply:
If you just add length, then you have 16 chars lowercase alphanumeric, should be enough entropy for anyone, no?

Reply:
But then I'd say we also need to change `genadminpassword` to NOT use A-Z, - and _, since then we have two different sets of characters to use for the same thing. But that also feels meh.

Reply:
I'm also fine with adding a $mode = easy/difficult instead of length, and vary charset and length based on it.

In any case, genadminpassword should just use this function (via the console command)?

Reply:
I can just try to use it there and see what happens.

Reply:
Wait but then we need a flag to the command like `--write-to-file` or something that also writes the password to the `etc/initial_admin_password` file

Reply:
Remove the file, say what command to run to set initial value? Or is that too complicated for users?

Reply:
or maybe only run the command for install of the db and not upgrade? The command shows the password so that should help I guess

Reply:
I like the password in a file, and it's used in [integration.sh](https://github.com/DOMjudge/domjudge/blob/2e35a3644984f8fd563fff4003e1beddcc27fb6a/gitlab/integration.sh#L44).

[This migration](https://github.com/DOMjudge/domjudge/blob/2e35a3644984f8fd563fff4003e1beddcc27fb6a/webapp/src/Migrations/Version20190803123217.php#L740) currently reads the file, it could be changed to generate the password and write it to the file. That would also work for restapi.secret. (I'm not sure about symfony_app.secret, which might need to exist earlier; by the way, its [recommended length](https://symfony.com/doc/current/reference/configuration/framework.html#secret) is around 32 chars.)

Reply:
Just thinking that maybe we’re creeping the scope here a bit much. So it might also be ok to just merge this with the slightly duplicated code. Then at least we have this useful feature. Not sure on second thought if changing the initial password generation is worth it or just premature optimization

Reply:
I agree, although I think we can still merge the functions. Propsal

* Add `$moreEntropy` to the generate function, update the docs to say that $moreEntropy=false is useful for when you are handing out the passwords on paper.
* If `$moreEntropy = true`, use length 16 and the expanded charset, otherwise not
* Update the tests

And then we can do other stuff later.

Reply:
Should be fixed now.